stdparams.go

```diff
@@ -0,0 +1,38 @@
+// +build go1.7
+
+package httprouter
+
+import (
+	"context"
+	"net/http"
+)
+
+type paramsKey struct{}
+
+// ParamsKey is the request context key under which URL params are stored.
+//
+// This is only present from go 1.7.
+var ParamsKey = paramsKey{}
+
+// Handler is an adapter which allows the usage of an http.Handler as a
+// request handle. With go 1.7+, the Params will be available in the
+// request context under ParamsKey.
+func (r *Router) Handler(method, path string, handler http.Handler) {
+	r.Handle(method, path,
+		func(w http.ResponseWriter, req *http.Request, p Params) {
+			ctx := req.Context()
+			ctx = context.WithValue(ctx, ParamsKey, p)
+			req = req.WithContext(ctx)
```

Comment:
We should update the README.md to reflect the additional allocations this change introduces (with which I would be totally fine with)

> **Zero Garbage:** The matching and dispatching process generates zero bytes of
> garbage. In fact, the only heap allocations that are made, is by building the
> slice of the key-value pairs for path parameters. If the request path contains
> no parameters, not a single heap allocation is necessary.